iclr2017/python/get-submission-notes.py

```diff
@@ -52,9 +57,71 @@
             paper_status[paper_num][id_str]['type'] = note_type
             paper_status[paper_num][id_str]['content'] = note.content
         else:
-            note_type = note.invitation.split('paper' + str(paper_num))[1]
-            print("%s %s MISSED %s for %s" % (paper_num, author, note_type,
-                                              paper_status[paper_num][id_str]['type']))
+            note_type = note.invitation.split('paper'+str(paper_num))[1]
+            print("%s %s MISSED %s for %s" %(paper_num,author, note_type,
+                                                           paper_status[paper_num][id_str]['type']))
+
+# in order to come up with one set of anonymous names even thought the same person might be
+# reviewer1 on paper 324 and reviewer2 on paper 444, we find all of the real names,
+# then convert them each into a number
+
+# profile_by_email[email_id] = ~name
+profile_by_email = {}
+def check_profile(member):
+
+    if member.startswith('~'):
+        return member
+
+    if '@' in member:
+        if member in profile_by_email:
+            return profile_by_email[member]
+
+        response = requests.get(client.baseurl+'/user/profile?email=' + member, headers = headers)
+        if 'profile' not in response.json():
+            profile_by_email[member] = member
+            return member
+        profile = response.json()['profile']
+        profile_by_email[member] = profile['id']
+        return profile_by_email[member]
+
+    return member
+
+# attach real name to the anonymized name for reviewers
+reviewers = {}
+for r in anon_reviewers.json():
+    reviewer_id = r['id']
+    members = r['members']
+    if members:
+        reviewers[reviewer_id] = check_profile(members[0])
+
+# attach real name to the anonymized name for area chairs
+# and convert email ids to names
+for r in area_chairs.json():
+    reviewer_id = r['id']
+    # only look at areachair1,2... not "areachairs"
+    if not reviewer_id.endswith('areachairs'):
+        members = r['members']
+        if members:
+            reviewers[reviewer_id] = check_profile(members[0])
+
+# make sure ALL names are in
+for paper_num in paper_status:
+    for note_id in paper_status[paper_num]:
+        if note_id != 'pdf':
+            name = paper_status[paper_num][note_id]['author']
+            if name not in reviewers:
+                reviewers[name] = name
+
+# now that we know the real ID of everyone, we hide all of the names....
+new_val = 0
+for r in reviewers:
+    name = reviewers[r]
+    if type(name) != int:
+        new_val += 1
+        # it would be nice to start from 'r' and move forward
+        for y in reviewers:
```

Comment:
doing something like this is easier when the object you're iterating over is an ordered list, instead of a dictionary. You'd have to turn this into a list somehow to do what you're describing - whether that means "enumerating" the dict, or some other way of doing it...

(I think if it's running in a reasonable amount of time, it's fine to leave it as is)